week06/manager_test.go

```diff
@@ -0,0 +1,122 @@
+package week06
+
+import (
+	"errors"
+	"testing"
+)
+
+func Test_Manager(t *testing.T) {
+	t.Parallel()
+
+	t.Run("product creation flow", func(t *testing.T) {
+		validProdList := []int{1, 3, 5, 7, 9}
+		noQtyProdList := []int{0}
+
+		table := []struct {
+			name        string
+			workerCount int
+			prodQtys    []int
+			expCpCount  int
+			expErr      error
+		}{
+			{
+				name:        "with valid products and workers",
+				workerCount: 2,
+				prodQtys:    validProdList,
+				expCpCount:  len(validProdList),
+				expErr:      nil,
+			}, {
+				name:        "with an invalid product list",
+				workerCount: 1,
+				prodQtys:    noQtyProdList,
+				expCpCount:  0,
+				expErr:      ErrInvalidQuantity(0),
+			}, {
+				name:        "with an invalid emp count",
+				workerCount: 0,
+				prodQtys:    validProdList,
+				expCpCount:  0,
+				expErr:      ErrInvalidEmployeeCount(0),
+			},
+		}
+
+		for _, tt := range table {
+			t.Run(tt.name, func(t *testing.T) {
+				m := NewManager()
+
+				var inventory []CompletedProduct
+				products := genProductList(t, tt.prodQtys...)
+
+				err := startUpWorkerPool(t, m, tt.workerCount)
+				if err != nil && err != tt.expErr {
+					t.Fatalf("got error %v, expected none", err)
+				}
+
+				if !errors.Is(tt.expErr, ErrInvalidEmployeeCount(tt.workerCount)) {
+					runProductionLine(t, m, products, &inventory)
+
+					select {
+					case err := <-m.Errors():
+						if err != tt.expErr {
+							t.Fatalf("got error %v, expected  error %v", err, tt.expErr)
+						}
+					case <-m.Done():
+					}
+				}
+
+				if len(inventory) != tt.expCpCount {
+					t.Fatalf("got %v products, expected %v products", len(inventory), len(products))
+				}
+				m.Stop()
+			})
+		}
+	})
+
+}
+
+// Helpers
+
+// Returns a list or products that need to be produced
+func genProductList(t testing.TB, qtys ...int) []Product {
+	t.Helper()
+	var ps []Product
+	for _, q := range qtys {
+		ps = append(ps, Product{Quantity: q})
+	}
+	return ps
+}
+
+// Spin up the employee pool based on the provided worker count
+func startUpWorkerPool(t testing.TB, m *Manager, n int) error {
+	t.Helper()
+
+	err := m.Start(n)
```

Comment:
You need to make sure these resources are cleaned up. Using the `Cleanup` method on `testing.T` you can ensure that regardless of what happens in the test itself, the resource will be cleaned by the go test runner.

```suggestion
	err := m.Start(n)
	t.Cleanup(m.Stop)
```